test/unit/org/apache/cassandra/cql3/CQLTester.java

```diff
@@ -2351,4 +2359,12 @@ public boolean equals(Object o)
                 && Objects.equal(password, u.password);
         }
     }
+
+    protected static <K,V> Map<K,V> buildInsertionOrderedMap(K k1, V v1, K k2, V v2)
```

Comment:
We can use the already existent `CQLTester#map(Object...)` instead of adding a new method.

Reply:
What do you think about moving the implementation of `map` in this (and fixing up the implementation to take an arbitrary number of parameters), and then delegating `map` to this. The reason I say this is that this method is explicit in its intention to build a specifically-ordered map (which we need for these tests), while `map` doesn't document any contract about the kind of map it produces, even though it happens to do what we need?

Reply:
Why don't we just have `CQLTester#map()` return a `LinkedHashMap`, codifying the behavior it already exhibits?

Reply:
I suppose, to me, unless the method name/comments/something explicitly say _why_ it's returning a LinkedHashMap, someone else could easily come along next year and go "huh, this new Java 22 thing would totally replace this method, so I may as well just use it instead - no reason we have to use a LinkedHashMap here anyway..." I suppose we could document that we expect the return value from `map` to be iterable in insertion order via code comments as well, but then you can never change `map` (which otherwise seems really generic and has no need for in-order iteration) without then adding this new method with a different behavior, and then figuring out where it would break tests. If we have a method (maybe rename this one to `inOrderMap`?) that `map` delegates to, if we ever want to touch `map` we can just split it out without breaking the original purpose of having the map be explicitly an in order traversable map?

Reply:
I'd be cool w/ just renaming `map()` -> `linkedHashMap()` (that returns a `LinkedHashMap`) and then adding back a `map()` method, so we can avoid renaming for dozens of existing call-sites.

Reply:
Did what @maedhroz suggested, and added a comment above `linkedHashMap()` to explain that some tests require in-order enumeration of maps so it make sense to someone who sees the two and thinks "I'll just inline this" in case we later want to, for example, add a `sortedHashMap` method or something else more specific.